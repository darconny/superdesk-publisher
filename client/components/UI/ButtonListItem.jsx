import React from 'react'
import PropTypes from 'prop-types'

const ButtonListItem = ({ onClick, label }) => {
    return (
        <div className="sd-list-item" onClick={onClick}>
            <div className="sd-list-item__border"></div>
            <div className="sd-list-item__column sd-list-item__column--grow sd-list-item__column--no-border">
                <div className="sd-list-item__row">
                    <span className="sd-overflow-ellipsis">{label}</span>
                </div>
            </div>
            <div className="sd-list-item__action-menu">
                <i className="icon-chevron-right-thin"></i>
            </div>
        </div>
    );
}

ButtonListItem.propTypes = {
    onClick: PropTypes.func,
    label: PropTypes.string
}

ButtonListItem.defaultProps = {
    label: 'Add',
}

export default ButtonListItem
