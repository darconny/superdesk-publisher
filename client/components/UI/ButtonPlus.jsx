import React from 'react'
import PropTypes from 'prop-types'

const ButtonPlus = ({ onClick }) => {
    return (
        <button className="btn btn--primary btn--icon-only-circle" onClick={onClick}>
            <i className="icon-plus-large"></i>
        </button>
    )
}

ButtonPlus.propTypes = {
    onClick: PropTypes.func
}

export default ButtonPlus