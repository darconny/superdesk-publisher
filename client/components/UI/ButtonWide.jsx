import React from 'react'
import PropTypes from 'prop-types'

const ButtonWide = ({ onClick, label }) => {
    return <button className="btn btn--hollow btn--expanded btn--primary" onClick={onClick}>{label}</button>
}

ButtonWide.propTypes = {
    onClick: PropTypes.func,
    label: PropTypes.string
}

ButtonWide.defaultProps = {
    label: 'Add',
}

export default ButtonWide