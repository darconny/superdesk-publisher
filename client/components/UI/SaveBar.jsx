import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

const SaveBar = ({ save, cancel, isDisabled }) => {
  return (
    <div className="sd-collapse-box__sliding-toolbar-wrapper">
      <div className="sd-collapse-box__sliding-toolbar">
        <div className="sliding-toolbar__inner" />
        <a className="btn btn--hollow btn--ui-dark" onClick={cancel}>
          Cancel
        </a>
        <button
          className={classNames("btn btn--primary", {
            "btn--disabled": isDisabled
          })}
          onClick={save}
          disabled={isDisabled}
          data-testid="save-button"
        >
          Save
        </button>
      </div>
    </div>
  );
};

SaveBar.propTypes = {
  save: PropTypes.func.isRequired,
  cancel: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool
};

SaveBar.defaultProps = {
  isDisabled: false
};

export default SaveBar;
