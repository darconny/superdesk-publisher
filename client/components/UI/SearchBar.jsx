import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

class SearchBar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false
    };
  }

  toggle = () => {
    this.setState({ open: !this.state.open }, () => {});
  };

  render() {
    let handlerStyle = {};
    let inputStyle = {};

    if (this.props.style === "dark") {
      handlerStyle = { borderRight: "1px solid rgba(0, 0, 0, 0.1)" };
      inputStyle = { color: "#ffffff" };
    }
    return (
      <div
        className={classNames("flat-searchbar", { extended: this.state.open })}
      >
        <div className="search-handler" style={handlerStyle}>
          <label
            for="search-input"
            className="trigger-icon"
            onClick={this.toggle}
          >
            <i className="icon-search" style={inputStyle} />
          </label>
          <input
            id="search-input"
            type="text"
            placeholder="Search"
            value={this.props.value}
            onChange={this.props.onChange}
            style={inputStyle}
          />
        </div>
      </div>
    );
  }
}

SearchBar.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  style: PropTypes.string
};

SearchBar.defaultProps = {
  style: "lights1"
};

export default SearchBar;
