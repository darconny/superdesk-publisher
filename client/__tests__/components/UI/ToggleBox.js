import React from 'react'
import {ToggleBox} from '../../../components/UI/ToggleBox'
import { render, fireEvent } from '@testing-library/react'

it('UI/ToggleBox renders correctly', () => {
   const {container} = render(<ToggleBox title="test" />)

   expect(container.firstChild).toMatchSnapshot()
})
