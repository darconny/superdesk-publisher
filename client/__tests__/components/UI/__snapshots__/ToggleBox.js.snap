// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/ToggleBox renders correctly 1`] = `
<div
  class="toggle-box"
>
  <div
    class="toggle-box__header"
  >
    <div
      class="toggle-box__chevron"
    >
      <i
        class="icon-chevron-right-thin"
      />
    </div>
    <div
      class="toggle-box__label"
    >
      test
    </div>
    <div
      class="toggle-box__line"
    />
  </div>
  <div
    class="toggle-box__content-wraper"
  >
    <div
      class="toggle-box__content"
    />
  </div>
</div>
`;
